src/components/protocol_handler/test/protocol_handler_tm_test.cc

```diff
@@ -414,24 +445,31 @@ TEST_F(ProtocolHandlerImplTest,
        StartSession_Unprotected_SessionObserverAccept) {
   AddConnection();
   const ServiceType start_service = kRpc;
+
+  TestAsyncWaiter waiter;
+  uint32_t times = 0;
   // Expect ConnectionHandler check
   EXPECT_CALL(
       session_observer_mock,
       OnSessionStartedCallback(
           connection_id, NEW_SESSION_ID, start_service, PROTECTION_OFF, _))
       .
       // Return sessions start success
-      WillOnce(Return(session_id));
+      WillOnce(DoAll(NotifyTestAsyncWaiter(&waiter), Return(session_id)));
+  times++;
 
   SetProtocolVersion2();
   // Expect send Ack
   EXPECT_CALL(transport_manager_mock,
               SendMessageToDevice(
                   ControlMessage(FRAME_DATA_START_SERVICE_ACK, PROTECTION_OFF)))
-      .WillOnce(Return(E_SUCCESS));
+      .WillOnce(DoAll(NotifyTestAsyncWaiter(&waiter), Return(E_SUCCESS)));
+  times++;
 
   SendControlMessage(
       PROTECTION_OFF, start_service, NEW_SESSION_ID, FRAME_DATA_START_SERVICE);
+
+  EXPECT_TRUE(waiter.WaitFor(times++, kAsyncExpectationsTimeout));
```

Comment:
@istoilovgithub Postfix `++` has not effect at this point of testing. Do you mean prefix version?